src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -32,11 +31,14 @@
 
     @Before
     public void setUp() {
-        productDao =new ArrayListProductDao(new ArrayList<>(Arrays.asList(product1,product2,product3)));
-
+        productDao =ArrayListProductDao.getInstance();
+        productDao.clearAll();
         when(product1.getId()).thenReturn(1L);
         when(product2.getId()).thenReturn(2L);
         when(product3.getId()).thenReturn(3L);
+        productDao.save(product1);
```

Comment:
Don't use save method to inject mocks. Write a protected setter method for product list and remove clearAll method